pilot/pkg/model/gateway.go

```diff
@@ -45,7 +50,8 @@ func MergeGateways(gateways ...Config) *MergedGateway {
 	servers := make(map[uint32][]*networking.Server)
 	tlsServers := make(map[uint32][]*networking.Server)
 	plaintextServers := make(map[uint32][]*networking.Server)
-	rdsRouteConfigNames := make(map[string][]*networking.Server)
+	serversByRouteName := make(map[string][]*networking.Server)
+	routeNamesByServer := make(map[*networking.Server]string)
```

Comment:
are you sure this pointer based key thing is safe/deterministic and doesn't cause correctness issues?

Reply:
Yes, Go pointers as map keys are fully supported and guaranteed to be deterministic even with moving GCs (I'll try to find the source) – used in Go standard library, language spec examples. Also I found multiple occurrences thoughout Istio codebase. 

I tried both approaches initially, pointer version is cleaner/easier to follow.